teps/0096-pipelines-v1-release.md

```diff
@@ -0,0 +1,138 @@
+---
+status: proposed
+title: Pipelines V1 Release
+creation-date: '2021-11-29'
+last-updated: '2021-11-29'
+authors:
+- '@lbernick'
+- '@jerop'
+- '@bobcatfish'
+- '@vdemeester'
+see-also:
+- TEP-0033
+---
+
+# TEP-0096: Pipelines V1 Release
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Background](#background)
+- [Proposal](#proposal)
+  - [V1 Stability and Deprecation Policy](#v1-stability-and-deprecation-policy)
+  - [API Definition](#api-definition)
+  - [Use Cases](#use-cases)
+- [References](#references)
+<!-- /toc -->
+
+## Summary
+
+Today, Tekton provides Alpha and Beta APIs.
+This TEP proposes a path to releasing Tekton Pipelines V1 to provide a stable API that users can rely on.
+This TEP documents the API definition and stability policy for Pipelines V1,
+defines the use cases we would like to support for Pipelines V1, and discusses features needed to support these use cases.
+
+## Motivation
+
+To become the industry standard, cloud-native CI/CD platform,
+Tekton must provide users with a stable API they can rely on.
+Some users may be waiting for V1 stability before deciding to use Tekton.
+
+### Goals
+
+- Define stability policy for Pipelines V1.
+- Define supported and unsupported use cases for Pipelines V1.
+- Define criteria for a feature to be "required" or "optional" for Pipelines V1.
+- Define what updates, if any, are needed to the [Pipelines API definition](https://github.com/tektoncd/pipeline/blob/main/api_compatibility_policy.md#what-is-the-api) before V1 release.
+
+### Non Goals
+
+- Define the stability or V1 plans of other Tekton project.
+- Define stability policies for aspects of Pipelines unrelated to the API, such as performance, dogfooding,
+release automation, and testing and debugging features.
```

Comment:
I am not clear what this alludes to. Apart from the release automation, most of these aspects can impact or be part of the stability policy, isn't it? For example, we might have to tweak the existing design if Tekton Pipelines is performing poor.

Reply:
The idea here was really to capture the non goals in the [original v1 doc](https://docs.google.com/document/d/1h9n0Lod0OiJ_sP2HK8Ms7N04aee5LW8xfz5yMGCFMIs/edit#), but it sounds like performance may be something we do want to consider explicitly for v1. @vdemeester I think you had a similar comment; thoughts on moving performance out of non goals? Priti, would you want this non goal removed entirely or just reworded?

Reply:
I think the original V1 doc described these as non-goals to not restrict V1 to include these aspects. They could be very well related to the stability policies of the API. I think we can drop it for now as a non-goal and capture it in the next iterations. The time we captured these non-goals in the original V1 doc v/s now, there has been many enhancements and updates done, we might have to revisit based on the latest status of the project.

Reply:
sounds good, I've removed this as a non goal.